new resource `azurerm_stack_hci_deployment_setting` (#25646)
hashicorp/terraform-provider-azurerm: website/docs/r/stack_hci_deployment_setting.html.markdown

variable "local_admin_password" {
  description = "The password of the local administrator account."
  sensitive   = true
  type        = string
}

variable "domain_admin_user" {
  description = "The username of the domain account."
  sensitive   = true
  type        = string
}

variable "domain_admin_password" {
  description = "The password of the domain account."
  sensitive   = true
  type        = string
}

variable "deployment_user" {
  sensitive   = true
  type        = string
  description = "The username for deployment user."
}

variable "deployment_user_password" {
  sensitive   = true
  type        = string
  description = "The password for deployment user."
}

locals {
  servers = [
    {
      name        = "AzSHOST1",
      ipv4Address = "192.168.1.12"
    },
    {
      name        = "AzSHOST2",
      ipv4Address = "192.168.1.13"
    }
  ]
  connection_roles = [
    "Azure Connected Machine Onboarding",
    "Azure Connected Machine Resource Administrator",
    "Azure Resource Bridge Deployment Role"
  ]
  machine_roles = [
    "Key Vault Secrets User",
    "Azure Connected Machine Resource Manager",
    "Azure Stack HCI Device Management Role",
    "Reader"
  ]
}

> data "azurerm_resource_group" "example" {

Author: note added

      bitlocker_data_volume_enabled   = true
      credential_guard_enabled        = true
      drift_control_enabled           = true
      drtm_protection_enabled         = true
      hvci_protection_enabled         = true
      side_channel_mitigation_enabled = true
      smb_cluster_encryption_enabled  = false
      smb_signing_enabled             = true
      wdac_enabled                    = true
    }

    storage {
      configuration_mode = "Express"
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

* `stack_hci_cluster_id` - (Required) The ID of the Azure Stack HCI cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

* `arc_resource_ids` - (Required) Specifies a list of IDs of Azure ARC machine resource to be part of cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

* `scale_unit` - (Required) One or more `scale_unit` blocks as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `version` - (Required) The deployment template version, possible value is `10.0.0.0`. Changing this forces a new Stack HCI Deployment Setting to be created.

---

> A `adapter_property_override` block supports the following:

Author: keep as `adapter_property_override` as offline discussed

* `jumbo_packet` - (Optional) The jumbo frame size of the adapter. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

* `network_direct` - (Optional) The network direct of the adapter. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

* `network_direct_technology` - (Optional) The network direct technology of the adapter. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

---

A `cluster` block supports the following:

> * `azure_service_endpoint` - (Required) For Azure blob service endpoint type, select either Default or Custom domain. If you selected Custom domain, enter the domain for the blob service in this format `core.windows.net`. Changing this forces a new Stack HCI Deployment Setting to be created.

Author: description updated

> * `cloud_account_name` - (Required) Specify the Azure Storage account name for cloud witness for your Azure Stack HCI cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

Author: fixed

> * `name` - (Required) The cluster name provided when preparing Active Directory. It must be 3-15 characters long and contain only letters, numbers and hyphens. Changing this forces a new Stack HCI Deployment Setting to be created.

Author: fixed

> * `witness_path` - (Required) Specify the fileshare path for the local witness for your Azure Stack HCI cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

Author: fixed

> * `witness_type` - (Required) Use a cloud witness if you have internet access and if you use an Azure Storage account to provide a vote on cluster quorum. A cloud witness uses Azure Blob Storage to read or write a blob file and then uses it to arbitrate in split-brain resolution. Possible values are `Cloud`, `FileShare`. Changing this forces a new Stack HCI Deployment Setting to be created.

Author: fixed

---

A `host_network` block supports the following:

* `intent` - (Required) One or more `intent` blocks as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `storage_network` - (Required) One or more `storage_network` blocks as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `storage_auto_ip_enabled` - (Optional) Whether allows users to specify IPs and Mask for Storage NICs when Network ATC is not assigning the IPs for storage automatically. Optional parameter required only for 3 Nodes Switchless deployments. Possible values are `true` and `false`. Defaults to `true`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: what does it mean for `Optional parameter required only for 3 Nodes Switchless deployments`?

> * `storage_connectivity_switchless` - (Optional) Defines how the storage adapters between nodes are connected either switch or switch less. Possible values are `true` and `false`. Defaults to `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: should be named as `xxx_enabled`

Author: fixed

---

A `infrastructure_network` block supports the following:

* `dns_server` - (Required) Specifies a list of IPv4 addresses of the DNS servers in your environment. Changing this forces a new Stack HCI Deployment Setting to be created.

* `gateway` - (Required) Specifies the default gateway that should be used for the provided IP address space. It should be in the format of an IPv4 IP address. Changing this forces a new Stack HCI Deployment Setting to be created.

* `ip_pool` - (Required) One or more `ip_pool` blocks as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `subnet_mask` - (Required) Specifies the subnet mask that matches the provided IP address space. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `dhcp_enabled` - (Optional) Whether allows customers to use DHCP for Hosts and Cluster IPs. If disabled, the deployment will Defaults to static IPs. If enabled, gateway and DNS servers are not required. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `dhcp_enabled` - (Optional) Whether DHCP is enabled for Hosts and Cluster IPs. Default to `xxx`. Changing this forces a new Stack HCI Deployment Setting to be created.

-> **NOTE:** If set to `false`, the deployment will use static IPs. If set to `true`, the gateway and DNS servers are not required.  
```

Author: fixed

---

A `intent` block supports the following:

* `name` - (Required) Specifies the name of the intent. Changing this forces a new Stack HCI Deployment Setting to be created.

* `adapter` - (Required) Specifies a list of ID of network interfaces used for the network intent. Changing this forces a new Stack HCI Deployment Setting to be created.

* `traffic_type` - (Required) Specifies a list of network traffic types. Possible values are `Compute`, `Storage`, `Management`. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `adapter_property_override` - (Optional) A `adapter_property_override` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `overridden_adapter_property` - (Optional) A `adapter_property_override` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: keep as `adapter_property_override` as offline discussed

* `qos_policy_override` - (Optional) A `qos_policy_override` block as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `virtual_switch_configuration_override` - (Optional) A `virtual_switch_configuration_override` block as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `override_adapter_property_enabled` - (Optional) This parameter should only be modified based on your OEM guidance. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `adapter_property_overriding_enabled` - (Optional) Whether the overridden adapter properties are enabled. Default to `xxx`. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed. the name is changed to `adapter_property_override_enabled`

> * `override_qos_policy_enabled` - (Optional) This parameter should only be modified based on your OEM guidance. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `qos_policy_overriding_enabled` - (Optional) Whether the overridden qos policies are enabled. Default to `xxx`. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed. the name is changed to `qos_policy_override_enabled`

> * `override_virtual_switch_configuration_enabled` - (Optional) This parameter should only be modified based on your OEM guidance. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `override_virtual_switch_configuration_enabled` - (Optional) Whether overridden virtual switch configurations are enabled. Default to `xxx`. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed. the name is changed to `virtual_switch_configuration_override_enabled`

---

A `ip_pool` block supports the following:

* `ending_address` - (Required) Specifies starting IP address for the management network. A minimum of six free, contiguous IPv4 addresses (excluding your host IPs) are needed for infrastructure services such as clustering. Changing this forces a new Stack HCI Deployment Setting to be created.

* `starting_address` - (Required) Specifies ending IP address for the management network. A minimum of six free, contiguous IPv4 addresses (excluding your host IPs) are needed for infrastructure services such as clustering. Changing this forces a new Stack HCI Deployment Setting to be created.

---

A `observability` block supports the following:

> * `episodic_data_upload_enabled` - (Required) Whether to collect log data to facilitate quicker issue resolution. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: seems should be named as `data_logging_enabled`?

Author: can we keep this same to the swagger for more consistency.

> * `eu_location_enabled` - (Required) Whether to store data sent to Microsoft in EU. The log and diagnostic data is sent to the appropriate diagnostics servers depending upon where your cluster resides. Setting this to `false` results in all data sent to Microsoft to be stored outside of the EU. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: seems it is used to control: whether the log and diagnostic data should be sent to Microsoft in EU or outside of EU?

Author: updated

> * `streaming_data_client_enabled` - (Required) Whether send telemetry data to Microsoft. Possible values are `true` and `false`. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `streaming_data_client_enabled` - (Required) Whether the telemetry data will be sent to Microsoft. Defaults to `xxx`. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed

---

A `optional_service` block supports the following:

* `custom_location` - (Required) Specifies the name of custom location. A custom location will be created after the deployment is completed. Changing this forces a new Stack HCI Deployment Setting to be created.

---

A `physical_node` block supports the following:

* `ipv4_address` - (Required) Specifies the IPv4 address assigned to each physical server on your Azure Stack HCI cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

* `name` - (Required) The NETBIOS name of each physical server on your Azure Stack HCI cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

---

A `qos_policy_override` block supports the following:

> * `bandwidth_percentage_smb` - (Optional) Specifies the bandwidth allocation in % for the storage traffic. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `bandwidth_percentage_smb` - (Optional) Specifies the percentage of the allocated storage traffic bandwidth. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed

* `priority_value8021_action_cluster` - (Optional) Specifies the Cluster traffic priority. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

* `priority_value8021_action_smb` - (Optional) Specifies the Priority Flow Control where Data Center Bridging (DCB) is used. This parameter should only be modified based on your OEM guidance. Changing this forces a new Stack HCI Deployment Setting to be created.

---

A `scale_unit` block supports the following:

* `adou_path` - (Required) Specify the full name of the Active Directory Organizational Unit container object prepared for the deployment, including the domain components. For example:`OU=HCI01,DC=contoso,DC=com`. Changing this forces a new Stack HCI Deployment Setting to be created.

* `cluster` - (Required) A `cluster` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `domain_fqdn` - (Required) Specifies the FQDN to deploy cluster. Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: ```suggestion
* `domain_fqdn` - (Required) Specifies the FQDN for deploying the cluster. Changing this forces a new Stack HCI Deployment Setting to be created.
```

Author: fixed

* `host_network` - (Required) A `host_network` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

* `infrastructure_network` - (Required) One or more `infrastructure_network` blocks as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

> * `naming_prefix` - (Required) Specifies the naming prefix to deploy cluster. It must be 1-8 characters long and contain only letters, numbers and hyphens Changing this forces a new Stack HCI Deployment Setting to be created.

Reviewer: is it the name prefix of the cluster or something else?

Author: this is not the cluster naming prefix, it is used in naming somthing produced during deployment.

Author: changed to `name_prefix`

* `optional_service` - (Required) A `optional_service` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

* `physical_node` - (Required) One or more `physical_node` blocks as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

* `secrets_location` - (Required) The URI to the keyvault or secret store. Changing this forces a new Stack HCI Deployment Setting to be created.

* `security_setting` - (Required) A `security_setting` block as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `storage` - (Required) A `storage` block as defined below. Changing this forces a new Stack HCI Deployment Setting to be created.

* `observability` - (Required) A `observability` block as defined above. Changing this forces a new Stack HCI Deployment Setting to be created.

---

> A `security_setting` block supports the following:

Reviewer: please confirm the default values for all the sub properties

Author: The `security_setting` block is flattened and all sub-properties has set default values.

---
subcategory: "Azure Stack HCI"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_stack_hci_deployment_setting"
description: |-
  Manages a Stack HCI Deployment Setting.
---

# azurerm_stack_hci_deployment_setting

> Manages a Stack HCI Deployment Setting.

Reviewer 2: Can we also get a note added about the additional resources created on behalf of the user in their resource group to help ensure they are aware of their existence and importance, and that Terraform will attempt to remove these on deletion or recreation of this resource?
